databroker/v2.py

```diff
@@ -120,15 +120,18 @@ def temp():
     """
     from databroker._drivers.msgpack import BlueskyMsgpackCatalog
     handler_registry = {}
-    # Let ophyd be an optional dependency.
-    # If it is not installed, then we clearly do not need its handler for this
-    # temporary data store.
+
+    from databroker.core import discover_handlers
+    handler_registry = discover_handlers()
+    # Let ophyd be an optional dependency.  If it is not installed, then we
+    # clearly do not need its handler for this temporary data store.
     try:
         import ophyd.sim
+        if not handler_registry:
+            handler_registry['NPY_SEQ'] = ophyd.sim.NumpySeqHandler
```

Comment:
Why only register this if `handler_registry` is completely empty? If the intention is to avoid shadowing the `'NPY_SEQ'` item found by `discover_handlers()` (if any) should we check for that key specifically? Would `handler_registry.setdefault('NPY_SEQ', ophyd.sim.NumpySeqHandler)` be suitable?

Reply:
Yes, it was exactly the reason. I updated the code based on your suggestion.

Reply:
Great. Looks good. It’s worth noting for anyone following along that this special case is only necessary for ophyd < 1.4.0 because starting 1.4.0 ophyd declares its handler via an entrypoint.

Reply:
xref: https://github.com/bluesky/ophyd/pull/806